pkg/db/strategy.go

```diff
@@ -308,15 +309,27 @@ func (s *Strategy) getExisting(ctx context.Context, gvk schema.GroupVersionKind,
 }
 
 func (s *Strategy) Delete(ctx context.Context, obj types.Object) (types.Object, error) {
-	return s.Update(ctx, obj)
+	obj, err := s.Update(ctx, obj)
+	if err, ok := err.(*mysql.MySQLError); ok && err.Number == 1062 { // error 1062 is a duplicate entry error
+		return obj, newConflict(s.gvk, obj.GetName(), err)
+	}
```

Comment:
Could we do something like `err = translateDuplicateEntryErr(err)` for deduplication and to make it easier to have a "translation chain" here later (which we hopefully won't need)?
We have something similar here: https://github.com/acorn-io/runtime/blob/main/pkg/client/app.go#L240-L249

Reply:
Good call. Updated